zcash_primitives/src/sapling.rs

```diff
@@ -185,15 +185,36 @@ impl ProofGenerationKey {
     pub fn to_viewing_key(&self) -> ViewingKey {
         ViewingKey {
             ak: self.ak,
-            nk: constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk,
+            nk: NullifierDerivingKey(constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct NullifierDerivingKey(jubjub::SubgroupPoint);
+
+impl From<jubjub::SubgroupPoint> for NullifierDerivingKey {
+    fn from(p: jubjub::SubgroupPoint) -> Self {
+        NullifierDerivingKey(p)
+    }
+}
+
+impl From<NullifierDerivingKey> for jubjub::SubgroupPoint {
+    fn from(nk: NullifierDerivingKey) -> Self {
+        nk.0
+    }
+}
```

Comment:
I don't see this used anywhere. I presume it was added for parity with the other `From` impl, so if that one is removed then this likely also should be.

Reply:
It's used here: https://github.com/zcash/librustzcash/pull/589/commits/44b808d9cfe032b1ab3b5b2d4677a7f373b6b132#diff-65eea7ee46cbc6fdbcd35372d1907c9fe1c39f4e49e0cc760bd0662f25e0941bL284-R594

Reply:
I can just make the internal value crate-public, I guess. So nothing outside the crate can construct one of these.